install-usercss/install-usercss.js

```diff
@@ -391,47 +308,103 @@
+        if (e) isEnabled = e.target.checked;
+        if (installed || installedDup) {
+          (isEnabled ? start : stop)();
+          $('.install').disabled = isEnabled;
+          $('#live-reload-install-hint').hidden = !isEnabled;
+          $('#live-reload-install-hint-ff').hidden = !isEnabled || tabId < 0;
+        }
+      },
+    };
+    function check() {
+      getData(initialUrl)
+        .then(update, logError)
+        .then(() => {
+          timer = 0;
+          start();
+        });
+    }
+    function logError(error) {
+      console.warn(t('liveReloadError', error));
+    }
+    function start() {
+      timer = timer || setTimeout(check, DELAY);
+    }
+    function stop() {
+      clearTimeout(timer);
+      timer = 0;
+    }
+    function update(code) {
+      if (code == null) return;
+      sequence = sequence.then(() => {
+        const {id} = installed || installedDup;
+        const scrollInfo = cm.getScrollInfo();
+        const cursor = cm.getCursor();
+        cm.setValue(code);
+        cm.setCursor(cursor);
+        cm.scrollTo(scrollInfo.left, scrollInfo.top);
+        return API.installUsercss({id, sourceCode: code})
+          .then(updateMeta)
+          .catch(showError);
+      });
+    }
+    function DirectDownloader() {
+      let oldCode = null;
+      const wrapCode = code => {
+        const isSame = code === oldCode;
+        oldCode = code;
+        return isSame ? null : code;
+      };
+      return () => download(initialUrl).then(wrapCode);
+    }
+    function PortDownloader() {
+      const resolvers = new Map();
+      const port = chrome.tabs.connect(tabId, {name: 'downloadSelf'});
+      port.onMessage.addListener(({id, code, error}) => {
+        const r = resolvers.get(id);
+        resolvers.delete(id);
+        if (error) {
+          r.reject(error);
+        } else {
+          r.resolve(code);
+        }
+      });
+      port.onDisconnect.addListener(() => {
+        chrome.tabs.get(tabId, tab => {
+          if (chrome.runtime.lastError) {
+            closeCurrentTab();
+          } else if (tab.url === initialUrl) {
+            location.reload();
           }
-        };
-        chrome.storage.onChanged.addListener(waitForCodeInStorage);
-        return;
-      }
-
-      // on the off-chance dbExecChromeStorage.getAll ran right after bg download was saved
-      download(params.get('updateUrl'))
-        .then(initSourceCode)
-        .catch(err => messageBox.alert(t('styleInstallFailed', String(err)), 'pre'));
-    });
-  }
-
-  function onPortDisconnected() {
-    chrome.tabs.get(tabId, tab => {
-      if (chrome.runtime.lastError) {
-        closeCurrentTab();
-      } else if (tab.url === tabUrl) {
-        location.reload();
-      }
-    });
+        });
+      });
+      return () => new Promise((resolve, reject) => {
+        const id = performance.now();
+        resolvers.set(id, {resolve, reject});
+        port.postMessage({id, timer: true});
```

Comment:
If `timer` is always `true` then I think we can remove this parameter.

Reply:
Eek, it's a bug.